File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        if(nums.length<2) {
            return;
        }
        int slow = 0;
        int fast = 1;

        while(fast<nums.length) {
            if(nums[slow]!=0) {
                slow++;
                fast++;
            } else if(nums[fast]==0) {
                fast++;
            } else {
                int temp = nums[fast];
                nums[fast] = nums[slow];
                nums[slow] = temp;
            }
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = len(nums)
        if l < 2:
            return
        slow = 0
        fast = 1
        while fast < l:
            if nums[slow] != 0:
                slow+=1
                fast+=1
            elif nums[fast] == 0:
                fast+=1
            else:
                nums[slow],nums[fast] = nums[fast],nums[slow]
        return nums

        